src/services/services.ts

```diff
@@ -8342,6 +8441,15 @@ namespace ts {
         };
     }
 
+    function forEachDescendant(node: Node, kind: SyntaxKind, action: (node: Node) => void) {

```

Comment:
How about `forEachDescendantOfKind`? Want to make it clear that we're recursing.